### PR TITLE
Support sprite animation

### PR DESCRIPTION
**Pull request by luzhuang**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/galacean/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### Other information:

### CODE REVIEW

File: packages/core/src/animation/animationCurve/AnimationArrayCurve.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { IAnimationCurveCalculator } from "./interfaces/IAnimationCurveCalculato
 @StaticInterfaceImplement<IAnimationCurveCalculator<number[]>>()
 export class AnimationArrayCurve extends AnimationCurve<number[]> {
   /** @internal */
-  static _isReferenceType: boolean = true;
+  static _keepOriginReference: boolean = true;
```
Comment (GuoLei1990) [inline packages/core/src/animation/animationCurve/AnimationArrayCurve.ts:14]: `_keepOriginReference` is strange, the context is `AnimationArrayCurve`, so the name should `_isCopyMode`

File: packages/core/src/animation/internal/animationCurveOwner/AnimationCurveOwner.ts [hunk 2]
```diff
@@ -77,6 +77,10 @@ export class AnimationCurveOwner<V extends KeyframeValueType> {
     crossWeight: number,
     additive: boolean
   ): KeyframeValueType {
+    if (!this.cureType._isInterpolationType) {
+      return this.evaluateValue(destCurve, destTime, false);
+    }
```
Comment (GuoLei1990) [inline packages/core/src/animation/internal/animationCurveOwner/AnimationCurveOwner.ts:82]: Not consider additive mode, use `evaluateValue` directly?

File: packages/core/src/animation/internal/animationCurveOwner/AnimationCurveOwner.ts [hunk 3]
```diff
@@ -104,6 +108,10 @@ export class AnimationCurveOwner<V extends KeyframeValueType> {
...
```
Comment (GuoLei1990) [inline packages/core/src/animation/internal/animationCurveOwner/AnimationCurveOwner.ts:113]: Not consider additive mode, use evaluateValue directly?

File: packages/core/src/animation/animationCurve/interfaces/IAnimationCurveCalculator.ts [hunk 4]
```diff
@@ -6,16 +6,16 @@ import { Keyframe, KeyframeValueType } from "../../Keyframe";
...
-  _additiveValue(additive: V, weight: number, srcOut: V): V;
-  _subtractValue(src: V, base: V, out?: V): V;
-  _getZeroValue(out?: V): V;
+  _lerpValue?(src: V, dest: V, weight: number, out?: V): V;
+  _additiveValue?(additive: V, weight: number, srcOut: V): V;
+  _subtractValue?(src: V, base: V, out?: V): V;
+  _getZeroValue?(out?: V): V;
   _copyValue(src: V, out?: V): V;
-  _hermiteInterpolationValue(frame: Keyframe<V>, nextFrame: Keyframe<V>, t: number, dur: number, out?: V): V;
+  _hermiteInterpolationValue?(frame: Keyframe<V>, nextFrame: Keyframe<V>, t: number, dur: number, out?: V): V;
```
Comment (GuoLei1990) [inline packages/core/src/animation/animationCurve/interfaces/IAnimationCurveCalculator.ts:20]: Opt code:
```ts
/**
 * @internal
 */
export interface IAnimationCurveCalculator<V extends KeyframeValueType> {
  _isCopyMode: boolean;
  _supportInterpolationMode: boolean;
  _supportAdditiveMode: boolean;

  _initializeOwner(owner: AnimationCurveOwner<V>);
  _initializeLayerOwner(owner: AnimationCurveLayerOwner);

  // Copy need
  _setValue(src: V, out?: V): V;

  // Lerp need
  _lerpValue?(src: V, dest: V, weight: number, out?: V): V;
  _hermiteInterpolationValue?(frame: Keyframe<V>, nextFrame: Keyframe<V>, t: number, dur: number, out?: V): V;

  // Additive need
  _getZeroValue?(out?: V): V;
  _subtractValue?(src: V, base: V, out?: V): V;
  _additiveValue?(additive: V, weight: number, srcOut: V): V;
}
```

File: packages/core/src/animation/animationCurve/AnimationRectCurve.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  static _copyValue(source: Rect, out: Rect): Rect {
+    out.copyFrom(source);
+    return out;
+  }
+
+  constructor() {
+    super();
+    this._evaluateData.value = new Rect();
+  }
+}
```
Comment (GuoLei1990) [inline packages/core/src/animation/animationCurve/AnimationRectCurve.ts:47]: Not support additive mode?

File: packages/core/src/animation/internal/animationCurveOwner/AnimationCurveOwner.ts (hunk 2 above)
Comment (GuoLei1990) [reply packages/core/src/animation/internal/animationCurveOwner/AnimationCurveOwner.ts:82]: done

File: packages/core/src/animation/animationCurve/interfaces/IAnimationCurveCalculator.ts (hunk 4 above)
Comment (luzhuang) [reply packages/core/src/animation/animationCurve/interfaces/IAnimationCurveCalculator.ts:21]: done

File: packages/core/src/animation/animationCurve/AnimationArrayCurve.ts (hunk 1 above)
Comment (luzhuang) [reply packages/core/src/animation/animationCurve/AnimationArrayCurve.ts]: done

File: packages/core/src/animation/animationCurve/AnimationRectCurve.ts (hunk 5 above)
Comment (luzhuang) [reply packages/core/src/animation/animationCurve/AnimationRectCurve.ts:47]: yes, If there's any further need, we'll support it

**Review by GuoLei1990 — APPROVED** [review]

**Pull request merged**